lib/Service/SieveService.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @author Pierre Gordon <pierregordon@protonmail.com>
+ *
+ * Mail
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\Mail\Service;
+
+use Horde\ManageSieve;
+use OCA\Mail\Account;
+use OCA\Mail\Service\Sieve\Script;
+use OCA\Mail\Service\Sieve\ScriptFactory;
+
+class SieveService
+{
+
+	/** @var ManageSieve */
+	private $sieveClient;
+
+	/** @var ScriptFactory */
+	private $scriptFactory;
+
+	/**
+	 * FiltersService constructor.
+	 * @param ScriptFactory $scriptFactory
+	 */
+	public function __construct(ScriptFactory $scriptFactory)
+	{
+		$this->scriptFactory = $scriptFactory;
+	}
+
+	/**
+	 * @param Account $account
+	 * @return $this
+	 * @throws ManageSieve\Exception
+	 */
+	public function setAccount(Account $account): SieveService
```

Comment:
this is kind of an anti pattern. Please pass in the sieve client when the other methods are called